Raven.Tests.Issues/RavenDB_4561.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using Raven.Abstractions.Data;
+using Raven.Abstractions.Replication;
+using Raven.Client.Exceptions;
+using Raven.Json.Linq;
+using Raven.Tests.Common;
+using Raven.Tests.Core.Utils.Entities;
+using Xunit;
+
+namespace Raven.Tests.Issues
+{
+    public class RavenDB_4561 : ReplicationBase
+    {
+        [Fact]
+        public void CanProperlyMergeIncomingChangeWithConflictedDoc()
+        {
+            var store1 = CreateStore();
+            var store2 = CreateStore();
+
+            TellFirstInstanceToReplicateToSecondInstance();
+
+            // create doc on second node
+            using (var session = store2.OpenSession())
+            {
+                session.Store(new Company { Name = "second_1" }, Etag.Empty, "companies/1");
+                session.SaveChanges();
+            }
+
+            var etagAfterFirstSave = store2.DatabaseCommands.Get("companies/1").Etag;
+            
+            // create doc on first (it should create conflict)
+            using (var session = store1.OpenSession())
+            {
+                session.Store(new Company { Name = "first_1" }, Etag.Empty, "companies/1");
+                session.SaveChanges();
+            }
+
+            Etag etagWithFirstConflict = Etag.Empty;
+            try
+            {
+                WaitForReplication(store2, "companies/1", changedSince: etagAfterFirstSave);
+                Assert.False(true); // we expect conflict
+            }
+            catch (ConflictException e)
+            {
+                etagWithFirstConflict = e.Etag;
+            }
+
+            store2.DatabaseCommands.Put(Constants.RavenReplicationConfig, null, RavenJObject.FromObject(new ReplicationConfig
+            {
+                DocumentConflictResolution = StraightforwardConflictResolution.ResolveToLocal,
+                AttachmentConflictResolution = StraightforwardConflictResolution.ResolveToLocal
+            }), new RavenJObject());
+
+            // create doc on first (it should update conflicted document)
+            using (var session = store1.OpenSession())
+            {
+                session.Store(new Company { Name = "first_2" }, "companies/1");
+                session.SaveChanges();
+            }
+
+            for (int i = 0; i < RetriesCount; i++)
+            {
+                try
+                {
+                    store2.DatabaseCommands.Get("companies/1");
+                }
+                catch (ConflictException e)
+                {
+                    if (e.Etag != etagWithFirstConflict)
```

Comment:
Please have a look at my change in this PR in Plugins/LocalDocumentReplicationConflictResolver.cs:

If we have conflict locally, then we set ResolveToLocal and update document on other node, we have following to merge:
1) local conflict 
2) incoming document.

In such case, I think we should leave this document without any resolution. 

Of course when we have local document (without conflict) and set resolve to local, then it will pick local version.